gain_clik: 1
lambda_return: 1.0e-2
lambda_effort: 5.0e-2
clearance_threshold: 1.5
select_task_axis: [1, 1, 1, 0, 0, 0]
secondary_task: "clearance"

base_frame: "world"
tool_frame: "ur5_ee_link"
elbow_frame: "ur5_forearm_link"


# da mettere in ur5_on_guide
# ricorda di aggiungere il plugin


planner_ik_hi:
  type: "itia/control/TopicHwIfaceNodelet"
  joint_names:
  - ur5_shoulder_pan_joint
  - ur5_shoulder_lift_joint
  - ur5_elbow_joint
  - ur5_wrist_1_joint
  - ur5_wrist_2_joint
  - ur5_wrist_3_joint
  allowed_controller_types:
  - "controller_type1"
  feedback_joint_state_topic: "/ur5/joint_states"
  command_joint_state_topic: "/joint_redundancy_target"
  sampling_period: 0.008
  diagnostic_period: 0.1
  feedback_joint_state_timeout: 20

  spline_planner:
    type: "thor/PrefilterPosVelEffController"
    controlled_joint:
    - ur5_shoulder_pan_joint
    - ur5_shoulder_lift_joint
    - ur5_elbow_joint
    - ur5_wrist_1_joint
    - ur5_wrist_2_joint
    - ur5_wrist_3_joint
    spline_order: 2

redundancy_hw:
  type: "itia/control/TopicHwIfaceNodelet"
  joint_names:
  - ur5_shoulder_pan_joint
  - ur5_shoulder_lift_joint
  - ur5_elbow_joint
  - ur5_wrist_1_joint
  - ur5_wrist_3_joint
  - ur5_wrist_2_joint
  allowed_controller_types:
  - "controller_type1"
  feedback_joint_state_topic: "/ur5/joint_states"
  command_joint_state_topic: "/joint_pos_target"
  sampling_period: 0.008
  diagnostic_period: 0.1
  feedback_joint_state_timeout: 2

  ik_clearance:
    type: "inverse_kinematics_qp/InverseKinematicsQpPosVelEffController"

    gain_clik: 1
    lambda_return: 1.0e-2
    lambda_effort: 5.0e-2
    clearance_threshold: 1.5
    select_task_axis: [1, 1, 1, 0, 0, 0]
    secondary_task: "clearance"
    setpoint_topic_name: "/joint_redundancy_target"
